pkg/operator/controller/controller_router_deployment_test.go

```diff
@@ -394,11 +394,33 @@ func TestDeploymentConfigChanged(t *testing.T) {
 			},
 			expect: true,
 		},
+		{
+			description: "if a container pre-stop lifecycle handler is added",
+			mutate: func(deployment *appsv1.Deployment) {
+				deployment.Spec.Template.Spec.Containers[0].Lifecycle = &corev1.Lifecycle{
+					PreStop: &corev1.Handler{
+						Exec: &corev1.ExecAction{
+							Command: []string{"/var/lib/haproxy/shutdown-haproxy"},
+						},
+					},
+				}
+			},
+			expect: true,
+		},
+		{
+			description: "if the deployment terminationGracePeriodSeconds is changed",
+			mutate: func(deployment *appsv1.Deployment) {
+				ninety := int64(90)
+				deployment.Spec.Template.Spec.TerminationGracePeriodSeconds = &ninety
+			},
+			expect: true,
+		},
 	}
 
 	for _, tc := range testCases {
 		nineteen := int32(19)
 		fourTwenty := int32(420) // = 0644 octal.
+		thirty := int64(30)
```

Comment:
Is `thirty` expected to match the termination grace period of https://github.com/openshift/router/pull/28? If so, should this be documented here?

Reply:
I only chose it because it's the default value, which means that it is the value that the operator will see on an existing ingress controller's deployment when the operator is upgraded.  It really could be any value (including nil), as long as the "if the deployment terminationGracePeriodSeconds is changed" test case sets a distinct value.